MAID-2965: add log_or_panic macro

Fraser999 opened the pull request.

Fraser999 committed:
chore/all: minor cleanups

Fraser999 committed:
feat/log_or_panic: add log_or_panic macro

maidsafe-highfive commented:
r? @ustulation

(maidsafe_highfive has picked a reviewer for you, use r? to override)

Fraser999 committed:
chore/test: improve assertion failure message

Fraser999 committed:
tests/logging: change log tests to integration ones

This means each log test runs in its own unique process, so initialisation
of logging or expected log output doesn't conflict.
